spec/perimeter_x_cookie_validator_spec.rb

```diff
@@ -1,116 +1,116 @@
+    config = PxModule::Configuration.new(@params).configuration;
+    px_ctx = PxModule::PerimeterXContext.new(config, @req)
+    validator = PxModule::PerimeterxCookieValidator.new(config)
+
+    verified, px_ctx = validator.verify(px_ctx)
+    expect(verified).to eq true
+    expect(px_ctx.context[:blocking_reason]).to eq PxModule::COOKIE_HIGH_SCORE
+  end
+
+  it "verification failed on cookie validation" do
+    @req.cookies[:_px] = "B4+RFFKBmAUf8Tw7FKu913pI64IY+rok17S6taqatyVgNJyhUgPKt34SeXgjgNk5tTUD6EyafOnsRPUjnTm5gQ==:1000:aCCffFpX/TtFJBtloLoYGl83BF+wmzwPuEiqDU9aRDxIz46Tyo5nMTTZ737h74nbODI1kGhVZ3tz/NmR+MSkJZlOsc9wlbbvPVVOKU5qJqyI5OGeglG368ZWKZF+Cd+TQtoAIamUeNmNFxEBGqiGu9BW2EcsPDyWkWdXn2i1wvyWc/n6cl4LugG4+5P0bbMb"
+    config = PxModule::Configuration.new(@params).configuration;
+    px_ctx = PxModule::PerimeterXContext.new(config, @req)
+    validator = PxModule::PerimeterxCookieValidator.new(config)
+
+    verified, px_ctx = validator.verify(px_ctx)
+    expect(verified).to eq false
+    expect(px_ctx.context[:s2s_call_reason]).to eq PxModule::COOKIE_VALIDATION_FAILED
+  end
+
+  it "verification failed on cookie decryption fail and px_ctx should have px_orig_cookie" do
+    @req.cookies[:_px] = "B4+RFFKBmAdUf8Tw7FKu913pI64IY+rok17S6taqatyVgNJyhUgPKt34SeXgjgNk5tTUD6EyafOnsRPUjnTm5gQ==:1000:aCCffFpdX/TtFJBtloLoYGl83BF+wmzwPuEiqDU9aRDxIz46Tyo5nMTTZ737h74nbODI1kGhVZ3tz/NmR+MSkJZlOsc9wlbbvPVVOKU5qJqyI5OGeglG368ZWKZF+Cd+TQtoAIamUeNmNFxEBGqiGu9BW2EcsPDyWkWdXn2i1wvyWc/n6cl4LugG4+5P0bbMb"
+    config = PxModule::Configuration.new(@params).configuration;
+    px_ctx = PxModule::PerimeterXContext.new(config, @req)
+    validator = PxModule::PerimeterxCookieValidator.new(config)
+
+    verified, px_ctx = validator.verify(px_ctx)
+    expect(verified).to eq false
+    expect(px_ctx.context[:s2s_call_reason]).to eq PxModule::COOKIE_DECRYPTION_FAILED
+    expect(px_ctx.context[:px_orig_cookie]).to eq(px_ctx.context[:px_cookie])
+  end
+
+  it "verification passed succesfully" do
+    @req.cookies[:_px] = "kN5gv3OjmmzaQLuPtx7D2QHQgvzKgF2LvX/hKpipNGUR9AaCwwlPZLs0XXbAZxNb2b+iLPsEv0qpAtkamYxy6Q==:1000:gMqzmSVEOMDz6x1Nwc799ULXP/LBIOMsJZA7UuQ0Yj/4zVTT5LxwwySXP5264/Ub9k6CgcMM3587cE6Mr4S8PeFdVejI5d4hDQJC+9LTD+7mNhio8wVO5nIsnFOnMVO31dRfk9u+Xff030y34CYRTiqOjb5ENTRNGR1KDAeqSRY/y/bly7pJSfNAb6Viw8eK"
+    config = PxModule::Configuration.new(@params).configuration;
+    px_ctx = PxModule::PerimeterXContext.new(config, @req)
+    validator = PxModule::PerimeterxCookieValidator.new(config)
+
+    verified, px_ctx = validator.verify(px_ctx)
+    expect(px_ctx.context[:s2s_call_reason]).to be_nil
+    expect(verified).to eq true
+  end
+
+  it "verification passed succesfully but route is sensitive" do
```

Comment:
this test doesn't really check that "verification passed succesfully", because verified is false and s2s_call_reason can still be sensitive route even if verification fails no?

Reply:
No, verification has to pass for us to get to sensitive_route